rust/src/client.rs

```diff
@@ -268,7 +268,7 @@ pub(crate) fn running_in_container() -> bool {
 }
 
 pub(crate) fn ostree_path_exists() -> bool {
-    Path::new("/ostree/repo/").exists()
+    ostree_ext::container_utils::is_bare_split_xattrs().unwrap()
```

Comment:
OK right; this is a bit tricky.  The new API is fallible - it can return an error.

I think the correct thing here is to change this code like this:

```
pub(crate) fn ostree_path_exists() -> CxxResult<bool> {
    Ok(ostree_ext::container_utils::is_bare_split_xattrs()?)
}
```

Then on the C++ side, use `CXX_TRY(ostree_path_exists)`.  There are other examples in the code.

Reply:
The function name here should also likely change to be `is_bare_split_xattrs()` to match the stricter semantics of the new API.

Reply:
thanks, on it.

Reply:
we want to use CXX_TRY and not CXX_TRY_VAL? I see an example that seems like exactly what I am looking for I think. https://github.com/coreos/rpm-ostree/blob/5ae5a8e7140536e143924721c6baeb7638de392a/src/daemon/rpmostree-sysroot-core.cxx#L416

Reply:
that is what I used on the latest push.